src/Microsoft.AspNet.Mvc.Core/ModelBinding/SimpleTypeModelBinder.cs

```diff
@@ -21,38 +19,48 @@ public sealed class TypeConverterModelBinder : IModelBinder
             }
 
             var valueProviderResult = await bindingContext.ValueProvider.GetValueAsync(bindingContext.ModelName);
-            if (valueProviderResult == null)
+            if (valueProviderResult == ValueProviderResult.None)
             {
                 // no entry
                 return null;
             }
 
-            object newModel;
             bindingContext.ModelState.SetModelValue(bindingContext.ModelName, valueProviderResult);
+
             try
             {
-                newModel = valueProviderResult.ConvertTo(bindingContext.ModelType);
-                ModelBindingHelper.ReplaceEmptyStringWithNull(bindingContext.ModelMetadata, ref newModel);
+                var model = valueProviderResult.ConvertTo(bindingContext.ModelType);
+
+                if (bindingContext.ModelType == typeof(string))
```

Comment:
Got a polite reminder that I was being an idiot offline. This code is :100: percent fine.